src/robusta/core/sinks/robusta/robusta_sink.py

```diff
@@ -467,6 +474,24 @@ def __should_run_history(self) -> bool:
             logging.error("Failed to check run history condition", exc_info=True)
             return False
 
+    def __discovery_watchdog(self):
+        logging.info("Cluster discovery watchdog initialized")
+        is_initial_health_check = True
+        while self.__active:
+            # if the initial thread is stuck it will register as "Healthy"
+            initial_discovery_stuck = not is_initial_health_check and self.last_send_time == 0
+            # if is not healthy we will get the stack dump from sigint handler
+            if initial_discovery_stuck:
+                StackTracer.dump(traces=1)
+            if not self.is_healthy() or initial_discovery_stuck:
+                logging.info("Unhealthy discovery, restarting runner")
+                # sys.exit and thread.interrupt_main doest stop robusta
+                os.kill(os.getpid(), signal.SIGINT)
+                return
+            is_initial_health_check = False
+            time.sleep(DISCOVERY_CHECK_THRESHOLD_SEC)
```

Comment:
I think we need a shorter sleep period.
Other wise, the actual recovery time is 2*DISCOVERY_CHECK_THRESHOLD_SEC

Reply:
DISCOVERY_CHECK_THRESHOLD_SEC its default 90 minutes, 
I would make its own env var, how many minutes do you want the default to be?